packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx

```diff
@@ -117,304 +132,324 @@ export const dialogSlotClassNames: DialogSlotClassNames = {
-    overlay: {},
-    footer: {},
-    trapFocus: true,
-  };
+  const [open, setOpen] = useAutoControlled({
+    defaultValue: props.defaultOpen,
+    value: props.open,
+    initialValue: false,
+  });
 
-  static autoControlledProps = ['open'];
-  static Footer = DialogFooter;
+  const lockBodyScroll = () => {
+    const openDialogs = (+context.target.body.getAttribute(dialogsCounterAttribute) || 0) + 1;
+    context.target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);
 
-  actionHandlers = {
-    closeAndFocusTrigger: e => {
-      this.handleDialogCancel(e);
-      e.stopPropagation();
+    // Avoid to block scroll in nested dialogs
+    if (openDialogs === 1) {
+      disableBodyScroll(context.target.body);
+    }
+  };
 
-      _.invoke(this.triggerRef, 'current.focus');
-    },
-    close: e => this.handleDialogCancel(e),
+  const unlockBodyScroll = () => {
+    const openDialogs = (+context.target.body.getAttribute(dialogsCounterAttribute) || 0) - 1;
+    context.target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);
+
+    // Only enables scroll if all dialogs are closed
+    if (openDialogs === 0) {
+      enableBodyScroll(context.target.body);
+      context.target.body.removeAttribute(dialogsCounterAttribute);
+    }
   };
-  contentRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>;
-  overlayRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>;
-  triggerRef = React.createRef<HTMLElement>();
-
-  getInitialAutoControlledState(): DialogState {
-    return {
-      open: false,
-    };
-  }
-
-  static getAutoControlledStateFromProps(props: DialogProps, state: DialogState): Partial<DialogState> {
-    return {
-      contentId: getOrGenerateIdFromShorthand('dialog-content-', props.content, state.contentId),
-      headerId: getOrGenerateIdFromShorthand('dialog-header-', props.header, state.headerId),
-    };
-  }
-
-  handleDialogCancel = (e: Event | React.SyntheticEvent) => {
-    _.invoke(this.props, 'onCancel', e, { ...this.props, open: false });
-    this.setState({ open: false });
+
+  const handleDialogCancel = (e: Event | React.SyntheticEvent) => {
+    _.invoke(props, 'onCancel', e, { ...props, open: false });
+    setOpen(false);
+    unlockBodyScroll();
   };
 
-  handleDialogConfirm = (e: React.SyntheticEvent) => {
-    _.invoke(this.props, 'onConfirm', e, { ...this.props, open: false });
-    this.setState({ open: false });
+  const handleDialogConfirm = (e: React.SyntheticEvent) => {
+    _.invoke(props, 'onConfirm', e, { ...props, open: false });
+    setOpen(false);
+    unlockBodyScroll();
   };
 
-  handleDialogOpen = (e: React.SyntheticEvent) => {
-    _.invoke(this.props, 'onOpen', e, { ...this.props, open: true });
-    this.setState({ open: true });
+  const handleDialogOpen = (e: React.SyntheticEvent) => {
+    _.invoke(props, 'onOpen', e, { ...props, open: true });
+    setOpen(true);
+    lockBodyScroll();
```

Comment:
This should be under `useEffect`

Reply:
It didn't really under `useEffect`, previously in the componentDidUpdate we were comparing the prev and curr state, didn't worked quite the same with `useEffect`

Reply:
Just pushed inside the `useEffect` but to works as expected I had to set a flag to not go unlocking the body in the first render.